sdk/src/main/java/com/mapbox/maps/renderer/gl/TextureRenderer.kt

```diff
@@ -0,0 +1,155 @@
+package com.mapbox.maps.renderer.gl
+
+import android.opengl.GLES20
+import com.mapbox.maps.renderer.gl.GlUtils.toFloatBuffer
+
+internal class TextureRenderer(
+  private val depth: Float = 0f
+) {
+
+  private var program: Int = 0
+  private var attributePosition: Int = 0
+  private var attributeTextureCoord: Int = 0
+  private var uniformTexture: Int = 0
+  private var vbo: IntArray = IntArray(2)
+
+  fun initialize() {
+    setupVbo(
+      vertexArray = floatArrayOf(
+        -1f, -1f, depth,
+        1f, -1f, depth,
+        -1f, 1f, depth,
+        1f, 1f, depth,
+      ),
+      textureArray = floatArrayOf(
+        0f, 0f,
+        1f, 0f,
+        0f, 1f,
+        1f, 1f,
+      )
+    )
+
+    val vertexShader = GlUtils.loadShader(
+      GLES20.GL_VERTEX_SHADER,
+      VERTEX_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    val fragmentShader = GlUtils.loadShader(
+      GLES20.GL_FRAGMENT_SHADER,
+      FRAGMENT_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    program = GLES20.glCreateProgram().also {
+      GlUtils.checkError("glCreateProgram")
+      GLES20.glAttachShader(it, vertexShader).also { GlUtils.checkError("glAttachShader") }
```

Comment:
Why do we need `also` here? We can check error in the next line.

Reply:
Removed, indeed